component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/BeamProducerFinder.java

```diff
@@ -62,7 +65,8 @@ public Iterator<Record> find(final String familyName, final String inputName, fi
             log.warn("Component Kit Mapper instantiation failed, trying to wrap native beam mapper...");
             final Object delegate = Delegated.class.cast(mapper).getDelegate();
             if (PTransform.class.isInstance(delegate)) {
-                return new QueueInput(delegate, familyName, inputName, familyName, PTransform.class.cast(delegate));
+                QUEUE.add(new ArrayBlockingQueue<>(7, true));
```

Comment:
Why use _7_? (see me comment below on 10).
Maybe use the `CAPACITY` variable. See my comments on slack on performances.

Reply:
was one of my try, normally, it should not impact much performance; 
main thread consume the QUEUE while Beam Thread fill it.